sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpInstrumentationPolicy.java

```diff
@@ -203,37 +213,40 @@ public HttpInstrumentationPolicy(HttpInstrumentationOptions instrumentationOptio
     @SuppressWarnings("try")
     @Override
     public Response<?> process(HttpRequest request, HttpPipelineNextPolicy next) {
-        boolean isTracingEnabled = tracer.isEnabled();
-        if (!isTracingEnabled && !isLoggingEnabled) {
+        final boolean isTracingEnabled = tracer.isEnabled();
+        final boolean isMetricsEnabled = meter.isEnabled();
```

Comment:
Do these states ever change during the lifetime of the Tracer or Meter?

Reply:
not today. They might be dynamic some day. Let's make them final instance var for now - we can change it later.